From 29e58fa5df457d1f7cf7bedc73cbd12058fe075f Mon Sep 17 00:00:00 2001
From: Roberto Sartori <roberto.sartori.android@gmail.com>
Date: Wed, 4 Sep 2024 16:52:31 +0200
Subject: [PATCH] Remove duplicated 'gralloc_defaults'

Also add the header file from gralloc (copied from the gralloc1 directory
on this repo).

Signed-off-by: Roberto Sartori <roberto.sartori.android@gmail.com>
---
 Android.bp                       | 21 ---------------
 gralloc/private_handle_members.h | 45 ++++++++++++++++++++++++++++++++
 2 files changed, 45 insertions(+), 21 deletions(-)
 create mode 100644 gralloc/private_handle_members.h

diff --git a/Android.bp b/Android.bp
index 6eed8ddc..8b6997e7 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,26 +1,5 @@
-bootstrap_go_package {
-    name: "soong-gralloc_defaults",
-    pkgPath: "vendor/qcom/opensource/display-commonsys-intf",
-    deps: [
-        "blueprint",
-        "blueprint-pathtools",
-        "soong",
-        "soong-android",
-        "soong-cc",
-    ],
-    srcs: [
-        "gralloc_defaults.go",
-    ],
-    pluginFor: ["soong_build"],
-}
-
-gralloc_defaults {
-    name: "gralloc_defaults",
-}
-
 cc_library_headers {
     name: "display_intf_headers",
-    defaults: ["gralloc_defaults"],
     vendor_available: true,
     export_include_dirs: [
         "gralloc",
diff --git a/gralloc/private_handle_members.h b/gralloc/private_handle_members.h
new file mode 100644
index 00000000..137b2ea4
--- /dev/null
+++ b/gralloc/private_handle_members.h
@@ -0,0 +1,45 @@
+  unsigned int size;
+  unsigned int offset;
+  unsigned int offset_metadata;
+  uint64_t base;
+  uint64_t base_metadata;
+  uint64_t gpuaddr;
+
+  unsigned int layer_count;
+  uint64_t id;
+  uint64_t usage;
+
+#ifdef __cplusplus
+  static const int kNumFds = 2;
+  static const int kMagic = 'gmsm';
+
+  static inline int NumInts() {
+    return ((sizeof(private_handle_t) - sizeof(native_handle_t)) / sizeof(int)) - kNumFds;
+  }
+
+  private_handle_t(int fd, int meta_fd, int flags, int width, int height, int uw, int uh,
+                   int format, int buf_type, unsigned int size, uint64_t usage = 0)
+      : fd(fd),
+        fd_metadata(meta_fd),
+        magic(kMagic),
+        flags(flags),
+        width(width),
+        height(height),
+        unaligned_width(uw),
+        unaligned_height(uh),
+        format(format),
+        buffer_type(buf_type),
+        size(size),
+        offset(0),
+        offset_metadata(0),
+        base(0),
+        base_metadata(0),
+        gpuaddr(0),
+        layer_count(1),
+        id(0),
+        usage(usage) {
+    version = static_cast<int>(sizeof(native_handle));
+    numInts = NumInts();
+    numFds = kNumFds;
+  }
+#endif
